pkg/controller/release/strategy_executor.go

```diff
@@ -328,18 +297,20 @@ func genReleaseStrategyStateEnforcer(curr, succ *releaseInfo) PipelineStep {
 */
 
 func (e *StrategyExecutor) Execute() (bool, []StrategyPatch, []ReleaseStrategyStateTransition, error) {
-	// Validation step: ensuring all release specs are pointing to a
-	// meaningful strategy stage.
-	for _, relinfo := range []*releaseInfo{e.prev, e.curr, e.succ} {
-		if relinfo == nil {
-			continue
-		}
-		strategy := relinfo.release.Spec.Environment.Strategy
-		targetStep := relinfo.release.Spec.TargetStep
-		if targetStep >= int32(len(strategy.Steps)) {
-			err := fmt.Errorf("no step %d in strategy for Release %q",
-				targetStep, controller.MetaKey(relinfo.release))
-			return false, nil, nil, shippererrors.NewUnrecoverableError(err)
+	isHead, hasTail := e.succ == nil, e.prev != nil
+	hasIncumbent := e.prev != nil || e.succ != nil
+
+	// There is no really a point in making any changes until the successor
+	// has completed it's transition, therefore we're hoilding off and aborting
+	// the pipeline execution. An alternative to this approach could be to make
+	// an autonomous move purely based on the picture of the world. But due to
+	// the limited visilibility of what's happening to the successor (as it
+	// might be following it's successor) it could be that a preliminary action
+	// would create more noise than help really.
+	if !isHead {
+		if !releaseutil.ReleaseAchievedTargetStep(e.succ.release) {
+			return false, nil, nil, shippererrors.NewContenderStrategyIncompleteError(
```

Comment:
Bailing out here does sound like the right thing to do, but is it really an "error" though? It's very much expected behavior that a release will be processed when it's not the head, and its successor hasn't achieved its target step yet: it's basically what every rollout is. Thoughts?

Reply:
The error class was mainly introduced for the sake of bubbling up some visibility. I'd agree: there is nothing exceptional happening to the current release if it's successor is still progressing. Removed the error value from return values.